collector/datacollector/datacollector.go

```diff
@@ -20,24 +20,33 @@ package main
 import (
 	"context"
 	"flag"
+	"io/ioutil"
 	"net/http"
 	"time"
 
 	"github.com/golang/glog"
+	"github.com/google/certificate-transparency-go/x509"
+	"github.com/google/certificate-transparency-go/x509util"
+	"github.com/google/monologue/certgen"
 	"github.com/google/monologue/collector"
 	"github.com/google/monologue/ctlog"
 	"github.com/google/monologue/storage/print"
+	"github.com/google/trillian/crypto/keys/pem"
 )
 
 var (
 	getRootsPeriod = flag.Duration("get_roots_period", 0, "How regularly the monitor should get root certificates from the Log")
 	getSTHPeriod   = flag.Duration("get_sth_period", 0, "How regularly the monitor should get an STH from the Log")
+	addChainPeriod = flag.Duration("add_chain_period", 0, "How regularly the monitor should submit a (pre-)certificate to the Log")
```

Comment:
I feel like there's probably a sensible default for this, but the real blocker is whether we have what we need for signing? What would you think of putting a default value here, and making the condition at line 71 include whether we have a `certgen.CA` (or what it takes to make one)?

Reply:
I'm quite keen to have the default for all of the modules be set to 'off' (for now - a lot of this main will be re-vamped later on), *especially* the one in charge of writing to a Log.  And while yes, having the signing info is a blocker, later changes will make those parts be read from a db rather than be passed in as flags, so we will need some other way to specify that the write part should be on/off.  To keep it consistent with the flags for turning the get-sth and get-roots parts on/off I'd prefer to have this flag that controls it.  Setting the default period to 0 (i.e. off) for get-sth and get-roots, but then to something >0 for add-chain would be irritatingly inconsistent (to me, at least).